### PR TITLE
Flora Improvement

### PR DESCRIPTION
**Pull request by Alterist**
Ports /tg/ jungle trees
Ports some /tg/ code regarding chopping trees
Ports Xmas trees
General improvement regarding general flora, allowing for welders, flamers and explosions to destroy them
Adds Flamer/Tree interaction, allowing them to be set on fire and be destroyed by it(WIP, pls help)

### CODE REVIEW

File: code/game/objects/structures/flora.dm [hunk 1]
```diff
@@ -1,48 +1,200 @@
-//trees
+/obj/structure/flora
+	anchored = TRUE
+	var/health = 25
+	var/health_max = 25
+
```
Comment (spookydonut) [inline code/game/objects/structures/flora.dm]:
```suggestion
 var/on_fire = FALSE
```

File: code/game/objects/structures/flora.dm [hunk 2]
```diff
@@ -1,48 +1,200 @@
...
+		multiplier += 1
+
+/obj/structure/flora/update_icon()
+	return
+
+/obj/structure/flora/fire_act()
+	if(!disposed)
+		spawn(rand(100,175))
+			cdel(src)
+
```
Comment (spookydonut) [inline code/game/objects/structures/flora.dm]:
```suggestion
/obj/structure/flora/process()
 if(health <= 0)
  cdel(src)
 if(!on_fire)
  processing_objects.Remove(src)
 health -= somevalue
```

File: code/game/objects/structures/flora.dm [hunk 3]
```diff
@@ -1,48 +1,200 @@
...
+	var/multiplier = 1
+	if(W.damtype == "fire") //Burn damage deals extra
+		multiplier += 1
+
+/obj/structure/flora/update_icon()
+	return
+
+/obj/structure/flora/fire_act()
+	if(!disposed)
```
Comment (spookydonut) [inline code/game/objects/structures/flora.dm]:
```suggestion
	if(on_fire)
		return
	on_fire = TRUE
	processing_objects.Add(src)
```

File: code/game/objects/structures/flora.dm [hunk 4]
```diff
@@ -1,48 +1,200 @@
...
+	user.animation_attack_on(src)
+
+	var/multiplier = 1
+	if(W.damtype == "fire") //Burn damage deals extra
+		multiplier += 1
+
+/obj/structure/flora/update_icon()
+	return
+
+/obj/structure/flora/fire_act()
```
Comment (spookydonut) [inline code/game/objects/structures/flora.dm]:
```suggestion
/obj/structure/flora/flamer_fire_act()
```

File: code/game/objects/structures/flora.dm [hunk 5]
```diff
@@ -1,48 +1,200 @@
...
+	switch(severity)
+		if(1)
+			cdel(src)
+		if(2)
+			if(prob(70))
+				cdel(src)
+		if(3)
+			if(prob(50))
+				cdel(src)
+
```
Comment (spookydonut) [inline code/game/objects/structures/flora.dm]:
```suggestion
/obj/structure/flora/Dispose()
 processing_objects.Remove(src)
 ..()
```

File: code/game/objects/structures/flora.dm [hunk 6]
```diff
@@ -1,48 +1,210 @@
...
+			cdel(src)
+		if(2)
+			if(prob(70))
+				cdel(src)
+		if(3)
+			if(prob(50))
+				cdel(src)
+/obj/structure/flora/Dispose()
+ processing_objects.Remove(src)
+ ..()
```
Comment (Rohesie) [inline code/game/objects/structures/flora.dm]:
```suggestion
 return ..()
```

File: code/game/objects/structures/flora.dm [hunk 7]
```diff
@@ -1,48 +1,210 @@
...
+		if(2.0)
+			health -= (rand(140, 300))
+		if(3.0)
+			health -= (rand(50, 100))
+	healthcheck()
+	return
+
+/obj/structure/flora/tree/bullet_act(var/obj/item/projectile/Proj)
+	health -= Proj.damage * 0.5
+	..()
```
Comment (Rohesie) [inline code/game/objects/structures/flora.dm]:
```suggestion
	. = ..()
```

Comment (Alterist) [reply code/game/objects/structures/flora.dm]: done

File: code/game/objects/structures/flora.dm (hunk 6 above)
Comment (Alterist) [reply code/game/objects/structures/flora.dm]: done

File: code/game/objects/structures/flora.dm [hunk 8]
```diff
@@ -1,48 +1,219 @@
...
+		return 0
+
+	var/damage = W.force
+	if(W.w_class < 4 || !W.sharp || W.force < 20) //only big strong sharp weapon are adequate
+		damage *= 0.25
+
+	if(istype(W, /obj/item/tool/weldingtool))
+		var/obj/item/tool/weldingtool/WT = W
+
+		if(WT.remove_fuel(0))
```
Comment (Rohesie) [inline code/game/objects/structures/flora.dm]: No fuel cost? Eye protection not needed?

Comment (Alterist) [reply code/game/objects/structures/flora.dm]: i missed the fuelcost part, that'll be changed, about the eye damage, does attacking a mob with a welder hurt eyes?

**Pull request closed without merging**